crates/biome_js_analyze/src/lint/nursery/no_bitwise_operators.rs

```diff
@@ -0,0 +1,167 @@
+            AnyExpressionWithBitwise::JsAssignmentExpression(node) => node.operator_token().ok(),
+        }
+    }
+    fn exist_bitwise_op(&self) -> bool {
+        match self {
+            AnyExpressionWithBitwise::JsBinaryExpression(binary_expr) => {
+                binary_expr.operator().ok().is_some_and(|op| {
+                    matches!(
+                        op,
+                        JsBinaryOperator::BitwiseAnd
+                            | JsBinaryOperator::BitwiseOr
+                            | JsBinaryOperator::BitwiseXor
+                            | JsBinaryOperator::LeftShift
+                            | JsBinaryOperator::RightShift
+                            | JsBinaryOperator::UnsignedRightShift
+                    )
+                })
+            }
+            AnyExpressionWithBitwise::JsUnaryExpression(unary_expr) => unary_expr
+                .operator()
+                .ok()
+                .is_some_and(|op| op == JsUnaryOperator::BitwiseNot),
+            AnyExpressionWithBitwise::JsAssignmentExpression(assign_expr) => {
+                assign_expr.operator().ok().is_some_and(|op| {
+                    matches!(
+                        op,
+                        JsAssignmentOperator::BitwiseAndAssign
+                            | JsAssignmentOperator::BitwiseOrAssign
+                            | JsAssignmentOperator::BitwiseXorAssign
+                            | JsAssignmentOperator::LeftShiftAssign
+                            | JsAssignmentOperator::RightShiftAssign
+                            | JsAssignmentOperator::UnsignedRightShiftAssign
+                    )
+                })
+            }
+        }
+    }
+}
+
+impl Rule for NoBitwiseOperators {
+    type Query = Ast<AnyExpressionWithBitwise>;
+    type State = JsSyntaxToken;
+    type Signals = Option<Self::State>;
+    type Options = NoBitwiseOperatorsOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let expr = ctx.query();
+        if expr.exist_bitwise_op() {
+            let op = expr.operator_token()?;
+            if ctx
+                .options()
+                .allow
+                .iter()
+                .any(|op_str| op_str.as_ref() == op.text_trimmed())
+            {
+                return None;
+            }
+            return Some(op);
+        }
+        None
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        let op = state.text_trimmed();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! {
+                    "Don't use '"<Emphasis>{op}</Emphasis>"' operator."
+                },
+            )
+            .note(markup! {
+                "Did you mean "<Emphasis>"&&"</Emphasis> " or " <Emphasis>"||"</Emphasis> " instead?"
+            }),
```

Comment:
I'm not sure why the diagnostic changed. The previous version seemed very good to me, we just needed to **add** one more line 😅 

Can we add it back?

Reply:
I asked for it, but I think my intention was slightly misinterpreted :sweat_smile:
https://github.com/biomejs/biome/pull/5274#discussion_r1979749050

Reply:
So the reasonable diagnostic should be
```suggestion
                markup! {
                    "Unexpected use of '"<Emphasis>{op}</Emphasis>"'."
                },
            )
            .note(markup! {
                "Use standard math operations instead."
            }),
```
 Is this right? @ematipico @arendjr

Reply:
So my idea would be like this:

* If the user used `|`, present the following diagnostic: "Did you mean `||` instead? If you want to use the bitwise operator, consider suppressing this diagnostic."
* If the user user `&`, present the following diagnostic: "Did you mean `&&` instead? If you want to use the bitwise operator, consider suppressing this diagnostic."
* If the user used `^`, present the following diagnostic: "Did you mean `**` (exponentiation) instead? If you want to use the bitwise operator, consider suppressing this diagnostic." (This is a trick one, because if someone writes `^` chances are they meant to use exponentiation.)
* And so on.